modules/popup-stack/lib/PopupStack.ts

```diff
@@ -113,16 +113,78 @@ interface Stack {
   };
 }
 
+/**
+ * Get a deeply nested dot-notation path from an arbitrary object safely. Will return `undefined` if
+ * path does not exist. This function is not meant to be type-safe. Use with caution.
+ * @param obj Any object
+ * @param path dot-notation path of a deep property
+ */
+function get(obj: any, path: string): any {
```

Comment:
@NicholasBoll I assume you didn't use lodash's `get` because you didn't want to add the dependency for this module?

Reply:
Yes. The following code would actually compress better though (since spreads can explode in Babel for IE11 support):

```ts
function get(obj: any, path: string): any {
  const parts = path.split('.')
  const first = parts.splice(0, 1)[0];
  if (parts.length && obj[first]) {
    return get(obj[first], parts.join('.'));
  } else {
    return obj[first];
  }
}

function set(obj: any, path: string, value: any): any {
  const parts = path.split('.')
  const first = parts.splice(0, 1)[0];
  if (parts.length) {
    if (obj[first] === undefined) {
      obj[first] = {};
    }
    set(obj[first], parts.join('.'), value);
  } else {
    obj[first] = value;
  }
  return value;
}
```

What do you think?

Here's the difference in Babel's REPL:

With Spreads:
```js
function _toArray(arr) { return _arrayWithHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && Symbol.iterator in Object(iter)) return Array.from(iter); }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function set(obj, path, value) {
  var _path$split = path.split('.'),
      _path$split2 = _toArray(_path$split),
      first = _path$split2[0],
      rest = _path$split2.slice(1);

  if (rest.length) {
    if (obj[first] === undefined) {
      obj[first] = {};
    }

    set(obj[first], rest.join('.'), value);
  } else {
    obj[first] = value;
  }

  return value;
}
```

With `splice`:
```js
function get(obj, path) {
  var parts = path.split('.');
  var first = parts.splice(0, 1)[0];

  if (parts.length && obj[first]) {
    return get(obj[first], parts.join('.'));
  } else {
    return obj[first];
  }
}
```

Reply:
That compresses to like 400 bytes with no depencencies

Reply:
If the behavior works, I don't think anyone will ever look too closely at these functions so I'm not too worried about the perceived complexity between the two. In this case I'd probably learn towards `splice` if we're trying to keep this package super minimal 👍